### ISSUE TITLE
Cache detail activity on tablets

### ISSUE THREAD

**Issue by fm-sys** [cgeo/cgeo#14010]
### Tell us your idea!

c:geos UI isn't really designed for larger landscape devices currently. I suspect, that a navigation drawer would significantly improve the workflow for tablet users. If implemented, it would be visible only at larger screens, phone UI would stay as is.

### What solution would you suggest?

> On devices with 1240dp+ widths, present destinations in a permanently visible or dismissible navigation drawer. Assign hierarchy based on how frequently or quickly a user needs to move between destinations.

![grafik](https://user-images.githubusercontent.com/64581222/220296004-c0581bc1-e8e4-41d1-ac14-eee8127d30da.png)

### Is there an alternative?

Don't optimize c:geo for tablets.

@Lineflyer do you know from support if there are users using c:geo on a tablet?

### Additional Information

https://m2.material.io/components/navigation-drawer

**Comment by moving-bits** [comment]
Slightly off-topic, but before we dive into optimizing c:geo for tablets I propose to make it more usable in landscape mode on mobile phones. We do have a navigation rail already (which comes up instead of nav bottom bar in landscape mode), but still especially the action bar takes up too much space on those small devices.

One idea for this: 
- Put the menu symbols in a straight line on the right side (similar to navigation rail, but not covering the zoom symbols), 
- and put all three lines of text ("Offline map", "280 caches" and content of filter bar) into a single line at the top, that has the height of today's filter bar.

**Comment by fm-sys** [comment]
> I  propose to make it more usable in landscape mode on mobile phones

Actually I don't understand why someone should use his mobile in landscape mode despite for watching videos. The whole android OS is primarily designed for portrait layouts and other apps render ugly on small devices as well. 

Phones have always been portrait devices while tablets with larger screen are mostly landscape. 

Writing our own vertical actionbar is not worth the effort IMHO and would conflict with material guidelines. But we could use a smaller actionbar height when rendering on a small landscape screen.

**Comment by moving-bits** [comment]
> Actually I don't understand why someone should use his mobile in landscape mode despite for watching videos. The whole android OS is primarily designed for portrait layouts and other apps render ugly on small devices as well.

Mostly agree, > 95% of the time I'm using portrait mode myself. We do have requests on support, though, from time to time, regarding landscape mode.

**Comment by Lineflyer** [comment]
I agree...however the distinction between "phone" and "tablet" gets more and more undefined...large phone screens and foldables are more likely to be used in landscape.
Therefore optimizing landscape mode is still a thing and goes hand in hand with better tablet UI.

**Comment by fm-sys** [comment]
I agree that foldables are a valid use case we should keep in mind.

It's possible to provide different layout files depending on orientation and/or screen size. The challenge will be to find good value when to use which layout.

**Comment by moving-bits** [comment]
Coming back to one of your opening questions:

> @Lineflyer do you know from support if there are users using c:geo on a tablet?

Yes, we have users on support from time to time that use c:geo on their tablet (for easier planning, as more map space is visible).

No frequent pattern of specific problems, though, only a few of them having problems to install c:geo on their tablet in the first place, if their device does not support GPS (and therefore Play Store claiming c:geo not being compatible with their device).

**Comment by moving-bits** [comment]
Pursuing the "navigation rail" idea: 

Content of this navigation rail basically would be what bottom navigation line is today (but maybe in a different order)? Eg.:
- details
- description
- logs
- friends' logs
- images
- inventory
- waypoints
- variables

Content of the page would be the respective fragment. Navigation line at the bottom would be switched off.

**Comment by fm-sys** [comment]
According to material guidelines a navigation rail shall only be used for top level navigation. Navigation inside a sub activity may be done via a navigation drawer (see screenshot in initial post).

I'm not saying that these guidelines are unassailable, but it may be confusing for users if the same UI component is used for top- and sublevel navigation.

**Comment by moving-bits** [comment]
You are right, bad wording from my side - I meant navigation drawer.

**Comment by fm-sys** [comment]
Yes, all these component names are not really intuitive ;)
